archaius2-core/src/main/java/com/netflix/archaius/config/PrefixedViewConfig.java

```diff
@@ -52,33 +52,46 @@ public State(Config config, String prefix) {
             });
         }
     }
+
+    /** Listener to update the state of the PrefixedViewConfig on any changes in the source config. */
+    private static class PrefixedViewConfigListener extends DependentConfigListener<PrefixedViewConfig> {
+        private PrefixedViewConfigListener(PrefixedViewConfig pvc) {
+            super(pvc);
+        }
+
+        @Override
+        public void onSourceConfigAdded(PrefixedViewConfig pvc, Config config) {
+            pvc.updateState(config);
+        }
+
+        @Override
+        public void onSourceConfigRemoved(PrefixedViewConfig pvc, Config config) {
+            pvc.updateState(config);
+        }
+
+        @Override
+        public void onSourceConfigUpdated(PrefixedViewConfig pvc, Config config) {
+            pvc.updateState(config);
+        }
+
+        @Override
+        public void onSourceError(Throwable error, PrefixedViewConfig pvc, Config config) {
+        }
+    }
     
     public PrefixedViewConfig(final String prefix, final Config config) {
         this.config = config;
-        this.prefix = prefix.endsWith(".") ? prefix : prefix + ".";
```

Comment:
I'm confused here. Did you or did you not revert the bug fix?

Reply:
The previous code used "prefix" from the args instead of "this.prefix" in the anonymous class, meaning that "this.prefix" was never used (the version with the trailing dot added). In order to keep the behavior the same as before, we remove this logic (which is why we still have the prefix.length()+1) since now we actually start using this.prefix.